src/issue.cpp

```diff
@@ -144,6 +144,7 @@ static const std::map<Issue::ReferenceRule, std::vector<std::string>> ruleToInfo
     {Issue::ReferenceRule::RESET_RESET_VALUE, {"RESET_RESET_VALUE", "2.11", baseSpecificationUrl, "specB11"}},
     {Issue::ReferenceRule::MATH_MATHML, {"MATH_MATHML", "2.12.1", baseSpecificationUrl, "specB12"}},
     {Issue::ReferenceRule::MATH_CHILD, {"MATH_CHILD", "2.12.2", baseSpecificationUrl, "specB12"}},
+    {Issue::ReferenceRule::MATH_CI, {"MATH_CI", "2.12.3", baseSpecificationUrl, "specB12"}},
     {Issue::ReferenceRule::MATH_CI_VARIABLE_REF, {"MATH_CI_VARIABLE_REF", "2.12.3", baseSpecificationUrl, "specB12"}},
```

Comment:
As you can see, we already have `MATH_CI` which is used in case a `ci` element doesn't have an identifier as a child (see [here](https://github.com/agarny/libcellml/blob/issue1082/src/validator.cpp#L2109-L2111)). However, here, I would like to distinguish that `MATH_CI` case with the case where an identifier is provided, BUT it's not a variable that exists in the component. The MathML rule addresses the first case, but not really the latter one. So, not sure what to do here.

Reply:
the CellML spec doesn't make that distinction. The spec just says (in rule 2.12.3) that `The contents of a MathML ci element MUST be a valid variable reference`. Adding a new check on `ci` elements having exactly one non-comment child node is not checking that rule - so I would be in favour of not adding this check...or at least not linking it to rule 2.12.3.

Reply:
We definitely need that check, so that the analyser can safely make use of that non-comment child node (should a model be considered valid).

Otherwise, I have just realised that `MATH_CI_VARIABLE_REF` was already present and that I actually added `MATH_CI` (still for the reason given [above](https://github.com/cellml/libcellml/pull/1083#discussion_r1098154787)). So, I think it should be fine to use `MATH_CI_VARIABLE_REF` where I used `MATH_CI` (see [here](https://github.com/agarny/libcellml/blob/issue1082/src/validator.cpp#L2109-L2111)), and remove `MATH_CI`.

This is what I have done in commit b93075a5.

Reply:
Seems reasonable to me.